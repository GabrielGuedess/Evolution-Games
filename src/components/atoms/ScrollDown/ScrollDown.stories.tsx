import { Story, Meta } from '@storybook/react/types-6-0';

import { ScrollDown } from './ScrollDown';

export default {
  title: 'Atoms/ScrollDown',
  component: ScrollDown,
} as Meta;

export const Default: Story = () => <ScrollDown />;
