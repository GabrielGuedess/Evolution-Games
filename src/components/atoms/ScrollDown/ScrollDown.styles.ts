import styled, { keyframes, css } from 'styled-components';

const animationScroll = keyframes`
   0% {
    transform: translate(0, 0);
    opacity: 0;
  }
  40% {
    opacity: 1;
  }
  80% {
    transform: translate(0, 2rem);
    opacity: 0;
  }
  100% {
    opacity: 0;
  }
`;

export const Wrapper = styled.button`
  ${({ theme }) => css`
    position: absolute;
    inset: auto auto 3rem 3rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    cursor: pointer;
    z-index: ${theme.layers.base};

    div {
      position: relative;
      width: 2rem;
      height: 4rem;
      border: 0.2rem solid ${theme.colors.white};
      border-radius: 5rem;

      :before {
        content: '';
        position: absolute;
        left: 50%;
        top: 0.6rem;
        width: 0.6rem;
        height: 0.6rem;
        margin-left: -0.3rem;
        background-color: ${theme.colors.white};
        border-radius: 100%;
        animation: ${animationScroll} 2s infinite;
      }
    }
  `}
`;
