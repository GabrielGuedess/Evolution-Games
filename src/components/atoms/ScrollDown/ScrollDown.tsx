import * as S from './ScrollDown.styles';

export const ScrollDown = () => (
  <S.Wrapper aria-label="Scroll Button">
    <div />
  </S.Wrapper>
);
