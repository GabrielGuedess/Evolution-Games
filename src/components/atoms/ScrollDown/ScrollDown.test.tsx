import { renderWithProviders } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { ScrollDown } from './ScrollDown';

describe('<ScrollDown />', () => {
  it('should render the ScrollDown', () => {
    const { container } = renderWithProviders(<ScrollDown />);

    expect(screen.getByLabelText('Scroll Button')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
