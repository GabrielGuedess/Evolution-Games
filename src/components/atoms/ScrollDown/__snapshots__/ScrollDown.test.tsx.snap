// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<ScrollDown /> should render the ScrollDown 1'] = `
.c0 {
  position: absolute;
  -webkit-inset: auto auto 3rem 3rem;
  -ms-inset: auto auto 3rem 3rem;
  inset: auto auto 3rem 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  z-index: 10;
}

.c0 div {
  position: relative;
  width: 2rem;
  height: 4rem;
  border: 0.2rem solid #FFFFFF;
  border-radius: 5rem;
}

.c0 div:before {
  content: '';
  position: absolute;
  left: 50%;
  top: 0.6rem;
  width: 0.6rem;
  height: 0.6rem;
  margin-left: -0.3rem;
  background-color: #FFFFFF;
  border-radius: 100%;
  -webkit-animation: bPWYhV 2s infinite;
  animation: bPWYhV 2s infinite;
}

<button
  aria-label="Scroll Button"
  class="c0"
>
  <div />
</button>
`;
